src/Mod/RuleSkill.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2010-2020 OpenXcom Developers.
+ *
+ * This file is part of OpenXcom.
+ *
+ * OpenXcom is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * OpenXcom is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with OpenXcom.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "RuleSkill.h"
+
+namespace OpenXcom
+{
+
+RuleSkill::RuleSkill(const std::string &type) : _type(type), _targetMode(BA_NONE), _compatibleWeapons({}), _requiredBonus({}), _isPsiRequired(false), _checkHandsOnly(true), _scriptId(-1)
+{}
+
+/**
+ * Loads the skill from a YAML file.
+ * @param node YAML node.
+ * @param mod Mod for the skill.
+ */
+void RuleSkill::load(const YAML::Node &node, Mod *mod, const ModScript &parsers)
+{
+	if (const YAML::Node &parent = node["refNode"])
+	{
+		load(parent, mod, parsers);
+	}
+	_scriptId = node["scriptId"].as<int>(_scriptId);
```

Comment:
This will be hard to maintained, better add normal name and add tags where it will be managed by scripts (you will have multiple different `id` with this).